lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/params/ClassifierTrainParams.scala

```diff
@@ -21,9 +20,13 @@ case class ClassifierTrainParams(passThroughArgs: Option[String],
                                  objectiveParams: ObjectiveParams,
                                  seedParams: SeedParams,
                                  categoricalParams: CategoricalParams) extends BaseTrainParams {
+  var numClass: Int = 1
```

Comment:
Case classes shouldnt have mutable state, you can get around this by adding a setNumClass method that returns another case class

Reply:
can you give an example of what you mean by setNumClass() method that returns an object?  (setters in my experience do not return values).  Do you mean recreate the whole object after changing 1 property?